### PR TITLE
Create a cluster version of the OQ Engine on Docker

### LINKED ISSUE #78: Add template for Docker workers and kubernets
Provide a guideline and a template for a multi-node Docker setup.

This may be integrated with compose (easy) and kubernets (more complex). Check Swarm too.

### PR DESCRIPTION
**Pull request by daniviga**
Closes #78 

Cluster can be started via docker-compose or manually.  No changes are required on the Engine side.

peek 2018-01-18 19-33

Some changes will be required to the CI job to allow CD.

https://ci.openquake.org/job/builders/job/docker-builder/108/

### CODE REVIEW

**Comment by daniviga** [comment]
See https://github.com/gem/oq-builders/issues/88 for considerations and roadmap

File: oq-docker/conf/openquake.cfg [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+port = 5672
+user = openquake
+password = openquake
+vhost = openquake
+# This is where tasks will be enqueued.
+celery_queue = celery
+
+[dbserver]
+# enable multi_user if you have a multiple user installation
+multi_user = false
```
Comment (daniviga) [inline oq-docker/conf/openquake.cfg]: Check if should be `true`

File: oq-docker/scripts/oq-start.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# This is required to load a custom  local_settings.py when 'oq webui' is used.
+export PYTHONPATH=$HOME
+
+oq dbserver start &
+
+if [ "$LOCKDOWN" == "enable" ]; then
```
Comment (daniviga) [inline oq-docker/scripts/oq-start.sh]: s/enable/enabled/

File: oq-docker/scripts/celery-wait.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (daniviga) [inline oq-docker/scripts/celery-wait.sh]: /bin/bash

File: oq-docker/README.md [hunk 4]
```diff
@@ -4,37 +4,106 @@ Exposes the WebUI and a `oq` cli.
...
->$ sudo docker start myoqcontainer
->$ sudo docker stop myoqcontainer
+$ sudo docker start myoqcontainer
+$ sudo docker stop myoqcontainer
 ```
 
-## Custom build args
+### Authentication support
+
+Authentication support for the WebUI/API can be enabled passing the `LOCKDOWN=enable` environment variable to the Docker container:
```
Comment (nastasi-oq) [inline oq-docker/README.md]: s/enable/enabled

File: oq-docker/README.md [hunk 5]
```diff
@@ -4,37 +4,106 @@ Exposes the WebUI and a `oq` cli.
...
 
 ```bash
---build-arg oq_branch=master      ## oq-engine branch
---build-arg tools_branch=mater ## oq standalone tools branch
+$ sudo docker run -e LOCKDOWN=enable --name myoqcontainer -d -p 8800:8800 openquake/engine
```
Comment (nastasi-oq) [inline oq-docker/README.md]: s/enable/enabled

File: oq-docker/scripts/oq-start.sh (hunk 2 above)
Comment (nastasi-oq) [inline oq-docker/scripts/oq-start.sh]: s/enable/enabled

**Review by nastasi-oq — APPROVED** [review]
A part of LOCKDOWN=enabled and the multi-user settings.
LGTM

**Pull request merged**